data-prepper-plugin-schema/src/main/java/org/opensearch/dataprepper/schemas/PluginConfigsJsonSchemaConverter.java

```diff
@@ -0,0 +1,119 @@
+package org.opensearch.dataprepper.schemas;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.github.victools.jsonschema.generator.OptionPreset;
+import com.github.victools.jsonschema.generator.SchemaVersion;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.configuration.ConditionalRoute;
+import org.opensearch.dataprepper.model.processor.Processor;
+import org.opensearch.dataprepper.model.sink.Sink;
+import org.opensearch.dataprepper.model.source.Source;
+import org.reflections.Reflections;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class PluginConfigsJsonSchemaConverter {
+    private static final Logger LOG = LoggerFactory.getLogger(PluginConfigsJsonSchemaConverter.class);
+    static final String DOCUMENTATION_LINK_KEY = "documentation";
+    static final String PLUGIN_NAME_KEY = "name";
+    static final String PLUGIN_DOCUMENTATION_URL_FORMAT =
+            "https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/%s/%s/";
+    static final Map<Class<?>, String> PLUGIN_TYPE_TO_URI_PARAMETER_MAP = Map.of(
+            Source.class, "sources",
+            Processor.class, "processors",
+            ConditionalRoute.class, "processors",
+            Buffer.class, "buffers",
+            Sink.class, "sinks"
+    );
+    static final String CONDITIONAL_ROUTE_PROCESSOR_NAME = "routes";
+    static final Map<String, Class<?>> PLUGIN_TYPE_NAME_TO_CLASS_MAP = Map.of(
+            "source", Source.class,
```

Comment:
Let's try to avoid repeating constants. Can we make use of these? Maybe make some static constants?

https://github.com/opensearch-project/data-prepper/blob/e35b4eaf26af79d67834c0f183a846f501be4c16/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/configuration/PipelineModel.java#L29